File: isma-compiler/hsm-core/src/main/java/ru/nstu/isma/core/hsm/var/HMDerivativeEquation.java
package ru.nstu.isma.core.hsm.var;

import java.io.Serializable;

/**
 * Created by Bessonov Alex
 * Date: 24.10.13
 * Time: 23:31
 */
public class HMDerivativeEquation extends HMEquation implements Serializable {
    protected HMConst initial;

    public HMDerivativeEquation(String code) {
        this.code = code;
        initial = new HMUnnamedConst(0);
    }

    public HMConst getInitial() {
        return initial;
    }

    public Double getInitialValue() {
        if (initial == null)
            return 0d;

        if (initial.getValue() == null)
            return 0d;

        return initial.getValue();
    }

    public void setInitial(HMConst initial) {
        this.initial = initial;
    }

    public void setInitial(double initial) {
        this.initial = new HMUnnamedConst(initial);
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/var/HMDerivativeEquation.kt
package ru.nstu.isma.hsm.`var`

import java.io.Serializable

/**
 * Created by Bessonov Alex
 * Date: 24.10.13
 * Time: 23:31
 */
open class HMDerivativeEquation(code: String?) : HMEquation(), Serializable {
    var initial: HMConst?
    val initialValue: Double?
        get() {
            if (initial == null) {
                return 0.0
            }
            return if (initial?.value == null) {
                0.0
            } else {
                initial!!.value
            }
        }

    fun setInitial(initial: Double) {
        this.initial = HMUnnamedConst(initial)
    }

    init {
        this.code = code
        initial = HMUnnamedConst(0.0)
    }
}